src/cpp/session/modules/SessionDependencies.cpp

```diff
@@ -43,6 +45,8 @@ namespace session {
 
 namespace {
 
+module_context::WaitForMethodFunction s_waitForInstallFeatureDependencies;
```

Comment:
It seems like this is necessary so that `.rs.installFeatureDependencies` can be a blocking call... I'm wondering if it'd be better for .rs.installFeatureDependencies to be async so we don't have to worry about the failure modes (i.e. suppose a client gets disconnected and loses state, such that it never invokes the waitformethod and you have to kill your session to restore order).

Reply:
My main worry is if someone runs code like:

```
import matplotlib.pyplot as plt
# make a plot with matplotlib
plt.show()
```

that initial attempt will fail if `png` is not installed. That makes me believe the install needs to be synchronous.

Reply:
Of course, thinking again after morning coffee, the safest way to make this happen when would be something like:

```
deps <- .rs.collectFeatureDependencies("python_plots")
install.packages(deps)
```

so that we can then install packages synchronously here without worrying about those failure modes.